spec/imsc-hrm.html

```diff
@@ -244,9 +244,11 @@ <h2>Terms and Definitions</h2>
       Unless noted otherwise, this specification applies to an <a>IMSC Document Instance</a>.
     </p>
     <p>
-      A sequence of consecutive <dfn data-cite="ttml2#terms-intermediate-synchronic-document">Intermediate Synchronic Documents</dfn>
-      that <i>conforms to the Hypothetical Render Model</i> SHALL be processed without <a>error</a> by the HRM algorithm
-      specified at <a href="#hypothetical-render-model-general"></a>.
+      Given a sequence of consecutive <dfn data-cite="ttml2#terms-intermediate-synchronic-document">Intermediate
+      Synchronic Documents</dfn> generated using the <i>Intermediate Synchronic Document Construction</i> procedure
+      specified in [[!TTML2]] from one or more input <a>IMSC Document Instances</a>, the sequence of
+      <a>Intermediate Synchronic Documents</a> <i>conforms to the Hypothetical Render Model</i> SHALL be processed
+      without <a>error</a> by the HRM algorithm specified at <a href="#hypothetical-render-model-general"></a>.
```

Comment:
This still seems to be talking about conformance of a sequence of ISDs, but since we don't expect folk to distribute ISDs, it's the source document instance(s) that need to be assessed for conformance, not the ISDs. I was thinking of something more along the lines of:

```suggestion
A sequence of consecutive input <a>IMSC Document Instances</a>`
SHALL <i>conform to the Hypothetical Render Model</i> if,
after generating from them a sequence of
<a>Intermediate Synchronic Documents</a> using the
<i>Intermediate Synchronic Document Construction</i> procedure
specified in [[!TTML2]], that sequence of
<a>Intermediate Synchronic Documents</a> can processed
without <a>error</a> by the HRM algorithm specified at <a href="#hypothetical-render-model-general"></a>.
```

Reply:
@nigelmegitt It is not always the case that all ISDs from an input document are used, e.g. in the case of in/out points. What is your concern with specifying the conformance against a sequence of ISDs, wherever they come from?

Reply:
@palemieux Three concerns:

1. I don't think there's anywhere near as much utility in validating some intermediate processing artefact as in validating the document that is actually being exchanged.
2. From a testing perspective, I expect the test input stimuli to be IMSC document instances, not sequences of ISDs.
3. It's a potential interop concern that different implementations might generate different ISD sequences from the same input document, so I think that ISD generation itself should be within the normative scope of the spec, not outside it. Albeit that the specification is only by reference to the algorithm in TTML2.

Reply:
> From a testing perspective, I expect the test input stimuli to be IMSC document instances, not sequences of ISDs.

Sure. That is a statement about testing, not conformance.

More importantly, how do you modify your proposal to account for the fact that only temporal portions of input documents will be used.

Reply:
>Sure. That is a statement about testing, not conformance.

For the Implementation Report we need tests, and the tests have to check for conformance to the normative provisions in the specification. If the normative text is only about testing ISD sequences, then we need to create ISD sequences for those tests. I don't think that would be helpful.

>More importantly, how do you modify your proposal to account for the fact that only temporal portions of input documents will be used.

I don't think the proposal needs to be modified. Can you say more about the use case you have in mind? Is it something like the need to apply a temporal window to the ISDs generated, to deal with scenarios such as documents wrapped in fragmented MP4 segments for streaming?

Reply:
> If the normative text is only about testing ISD sequences, then we need to create ISD sequences for those tests. I don't think that would be helpful.

I do not see why we would need to deliver ISD as test artifacts since there is only way to create them according to TTML2 and we are not testing TTML2.

> s it something like the need to apply a temporal window to the ISDs generated, to deal with scenarios such as documents wrapped in fragmented MP4 segments for streaming?

Yes.

Also, the last ISD of a document extends to indefinite time.

Reply:
TTML2 doesn't say they can only be made from TTML documents though - it even has an explicit syntax for ISD Sequences and ISDs, and a reader could easily imagine that it's that syntax that is being validated.

> the last ISD of a document extends to indefinite time.

The end time of the last ISD isn't relevant in the HRM though?

Reply:
> The end time of the last ISD isn't relevant in the HRM though?

We do not want to feed the HRM infinite duration ISDs.

Reply:
As a data point, [imscHRM](https://github.com/sandflow/imscHRM) uses a simple JSON file to specify a sequence of IMSC documents with in and out points:

https://github.com/sandflow/imscHRM/blob/main/src/test/resources/ttml/sequence001/manifest.json

Reply:
> We do not want to feed the HRM infinite duration ISDs.

Every document has exactly one of these though, so we can simply state the rule for handling them, no? There's no drawing cost for the ISD after the last, infinite one, because it doesn't exist.

Reply:
Re the in and out points, and the real world issue that external clipping times can apply to the presentation interval of any IMSC document, I'd suggest that we state that when invoking the [[resolve timing]](https://www.w3.org/TR/ttml2/#procedure-resolve-timing) procedure the active time duration is restricted to the externally provided in and out times for each IMSC document. Alternatively we could post-fix the ISD sequence, but that feels less elegant to me.

Reply:
See https://github.com/w3c/imsc-hrm/pull/60/commits/0df2807667eb2b875a9abf7820fa778ad156df01